tests/chainer_tests/dataset_tests/test_dataset_mixin.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+
+from chainer import dataset
+from chainer import testing
+
+
+class SimpleDataset(dataset.DatasetMixin):
+
+    def __init__(self, values):
+        self.values = values
+
+    def __len__(self):
+        return len(self.values)
+
+    def get_example(self, i):
+        return self.values[i]
+
+
+class TestDatasetMixin(unittest.TestCase):
+
+    def test_getitem(self):
+        ds = SimpleDataset([1, 2, 3, 4, 5])
+        for i in range(len(ds.values)):
```

Comment:
I think it is permissible to use range here, because `len(ds.values)` is not so large.